### ISSUE TITLE
Nettle fails to build on Centos 5

### ISSUE THREAD

**Issue by cswoodruff** [noloader/Build-Scripts#10]
I'm trying to build openssh on Centos 5 32-bit and I'm getting an error building dependency `nettle-3.5.1`.  I'm seeing a number of errors referencing an undefined `__builtin_bswap64`.

```
gcc -g2 -O2 -fPIC -pthread -ggdb3 -Wno-pointer-sign -Wall -W   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -L.. -L/usr/local/lib -Wl,-R,'$ORIGIN/../lib' -Wl,-R,/usr/local/lib -Wl,--enable-new-dtags nettle-benchmark.o nettle-openssl.o ../getopt.o ../getopt1.o ../nettle-internal.o timing.o -lnettle -lrt -lgmp -ldl -lpthread -lm -lcrypto -o nettle-benchmark
gcc -g2 -O2 -fPIC -pthread -ggdb3 -Wno-pointer-sign -Wall -W   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -L.. -L/usr/local/lib -Wl,-R,'$ORIGIN/../lib' -Wl,-R,/usr/local/lib -Wl,--enable-new-dtags base16enc.o io.o \
	-lnettle -lgmp -ldl -lpthread -o base16enc
../libnettle.so: undefined reference to `__builtin_bswap64'
collect2: ld returned 1 exit status
gmake[1]: *** [nettle-benchmark] Error 1
gmake[1]: *** Waiting for unfinished jobs....
../libnettle.so: undefined reference to `__builtin_bswap64'
collect2: ld returned 1 exit status
gmake[1]: *** [base16enc] Error 1
gmake[1]: Leaving directory `/home/lri/Build-Scripts/nettle-3.5.1/examples'
gmake: *** [all] Error 2
Failed to build Nettle
```

Full build log ...
[build-nettle.log](https://github.com/noloader/Build-Scripts/files/4100898/build-nettle.log)

```
$ uname -a
Linux localhost.localdomain 2.6.18-398.el5 #1 SMP Tue Sep 16 20:51:48 EDT 2014 i686 i686 i386 GNU/Linux
```

**Comment by noloader** [comment]
Thanks @cswoodruff.

Let me ping Neils to see what he recommends.

**Comment by noloader** [comment]
Thanks again @cswoodruff.

Niels already patched the issue; see [Commit 9f4180d6f418](https://git.lysator.liu.se/nettle/nettle/commit/9f4180d6f418).

However, we don't want to `autoreconf`. Autotools is a mess on downlevel clients, and it will turn a small problem into a big problem rather quickly.

This worked for me on CentOS 6 32-bit.

```diff
--- ctr.c
+++ ctr.c
@@ -48,6 +48,10 @@
 
 #define MIN(a,b) (((a) < (b)) ? (a) : (b))
 
+#if !defined(_LP64) && !defined(__LP64__)
+# undef HAVE_BUILTIN_BSWAP64
+#endif
+
 static size_t
 ctr_fill (size_t block_size, uint8_t *ctr, size_t length, uint8_t *buffer)
 {

```
